indico/modules/events/registration/models/tags.py

```diff
@@ -0,0 +1,60 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from sqlalchemy.ext.declarative import declared_attr
+
+from indico.core.db import db
+
+
+class RegistrationTag(db.Model):
+    """Registration tag used to mark/filter registrations."""
+
+    __tablename__ = 'registration_tags'
+
+    @declared_attr
+    def __table_args__(cls):
+        return (db.Index('ix_uq_registration_tags_title_lower', db.func.lower(cls.title), unique=True),
```

Comment:
This index needs to include `cls.event_id` as well (have a look at the one in `EditingTag`). Otherwise tag names would need to be unique globally across all events...